File: saizad-mvvm/src/main/java/com/saizad/mvvm/ui/calendar/DateTimePickerFragment.java
package com.saizad.mvvm.ui.calendar;

import android.app.DatePickerDialog;
import android.app.Dialog;
import android.app.TimePickerDialog;
import android.os.Bundle;

import androidx.annotation.NonNull;

import org.jetbrains.annotations.NotNull;
import org.joda.time.DateTime;

public class DateTimePickerFragment extends BaseCalendarPickerFragment {

    private final DateTime dateTime;

    public DateTimePickerFragment(@NonNull DateSelectedListener dateSetListener) {
        this(new DateTime(), dateSetListener);
    }

    public DateTimePickerFragment(@NonNull DateTime dateTime, @NonNull DateSelectedListener dateSelectedListener) {
        super(dateSelectedListener);
        this.dateTime = dateTime;
    }

    @NotNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        return new DatePickerDialog(getActivity(), (view, year1, month1, dayOfMonth) -> {
            new TimePickerDialog(getActivity(), (view1, hourOfDay, minute) -> {
                final DateTime dateTime = DateTime.parse(year1 + "-" + (month1 + 1) + "-" + dayOfMonth);
                dateSelectedListener.selected(dateTime.withTimeAtStartOfDay().plusHours(hourOfDay).plusMinutes(minute));
            }, dateTime.getHourOfDay(), dateTime.minuteOfHour().get(), false).show();
        }, dateTime.getYear(), dateTime.getMonthOfYear() - 1, dateTime.getDayOfMonth());
    }
}

File: saizad-mvvm/src/main/java/com/saizad/mvvm/ui/calendar/DateTimePickerFragment.kt
package com.saizad.mvvm.ui.calendar

import android.app.DatePickerDialog
import android.app.DatePickerDialog.OnDateSetListener
import android.app.Dialog
import android.app.TimePickerDialog
import android.app.TimePickerDialog.OnTimeSetListener
import android.os.Bundle
import android.widget.DatePicker
import android.widget.TimePicker
import org.joda.time.DateTime

class DateTimePickerFragment(
    private val dateTime: DateTime = DateTime(),
    private val minDateTime: DateTime = DateTime(0),
    dateSelectedListener: (DateTime) -> Unit
) : BaseCalendarPickerFragment(dateSelectedListener) {


    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val datePickerDialog = DatePickerDialog(
            requireContext(),
            OnDateSetListener { _, year, month, dayOfMonth ->
                TimePickerDialog(
                    activity,
                    OnTimeSetListener { view1: TimePicker?, hourOfDay: Int, minute: Int ->
                        val dateTime =
                            DateTime.parse(year.toString() + "-" + (month + 1) + "-" + dayOfMonth)
                        dateSelectedListener.invoke(
                            dateTime.withTimeAtStartOfDay().plusHours(hourOfDay).plusMinutes(minute)
                        )
                    },
                    dateTime.hourOfDay,
                    dateTime.minuteOfHour().get(),
                    false
                ).show()
            },
            dateTime.year,
            dateTime.monthOfYear - 1,
            dateTime.dayOfMonth
        )
        datePickerDialog.datePicker.minDate = minDateTime.millis
        return datePickerDialog
    }

}